packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -144,6 +149,21 @@ abstract class Route<T> {
   RouteSettings get settings => _settings;
   RouteSettings _settings;
 
+  /// The restoration scope ID to be used for the [RestorationScope] surrounding
+  /// this route.
+  ///
+  /// The restoration scope ID is null if restoration is currently disabled
+  /// for this route.
+  ///
+  /// If the restoration scope ID changes (e.g. because restoration is enabled
+  /// or disabled) throughout the life of the route, the [ValueListenable]
+  /// notifies its listeners. As an example, the ID changes to null while the
+  /// route is transitioning off screen. At that point, the route is considered
```

Comment:
"...transitioning off screen, which triggers a notification on this field" or some such. (Right now, you say "as an example" but don't actually then say why what you describe is an example of the previous sentence)